curriculum/challenges/english/25-front-end-development/workshop-merge-sort/656702f8b4cbd8cbf0a433c6.md

```diff
@@ -0,0 +1,54 @@
+---
+id: 656702f8b4cbd8cbf0a433c6
+title: Step 11
+challengeType: 20
+dashedName: step-11
+---
+
+# --description--
+
+Inside your function, create a `while` loop that compares an element in `left_part` to an element in `right_part`, and merges the smaller element to the main `array` list. 
+
+Create two conditions for the loop: one that checks whether the `left_array_index` is less than the length of `left_part` and another condition that checks whether `right_array_index` is less than the length of `right_part`.
+
+# --hints--
+
+Your `while` loop should have these conditions: `left_array_index < len(left_part) and right_array_index < len(right_part)`.
```

Comment:
Not sure if at this point the hint should contain the actual code to write in order to pass. Anyway, here we should also accept the loop with the conditions swapped.

Reply:
Modified the hint and added another test for swapped conditions.